Objective-C/Tests/ReplicatorTest+CustomConflict.m

```diff
@@ -233,6 +233,41 @@ - (void) testConflictResolverMergeDoc {
     AssertEqualObjects(savedDoc.toDictionary, exp);
 }
 
+- (void) testNonBlockingConflictResolver {
+    XCTestExpectation* ex = [self expectationWithDescription: @"testNonBlockingConflictResolver"];
+    NSDictionary* localData = @{@"key1": @"value1"};
+    NSDictionary* remoteData = @{@"key2": @"value2"};
+    [self makeConflictFor: @"doc1" withLocal: localData withRemote: remoteData];
+    [self makeConflictFor: @"doc2" withLocal: localData withRemote: remoteData];
+    
+    TestConflictResolver* resolver;
+    CBLReplicatorConfiguration* pullConfig = [self pullConfig];
+    
+    NSMutableArray* order = [NSMutableArray array];
+    resolver = [[TestConflictResolver alloc] initWithResolver: ^CBLDocument* (CBLConflict* con) {
+        @synchronized (order) {
+            [order addObject: con.localDocument.id];
+        }
+        if (order.count == 1) {
+            [NSThread sleepForTimeInterval: 0.5];
+            [ex fulfill];
+        }
+        @synchronized (order) {
+            [order addObject: con.localDocument.id];
```

Comment:
Could there be a race b/w adding the docID here and checking order assertion after the expectation is fulfill?

Reply:
you are right, there is no need for this. But now i wonder, whether we should fulFill the expectation after adding the last docID.

Reply:
udpated